projects/js-packages/charts/src/components/bar-list-chart/bar-list-chart.tsx

```diff
@@ -292,4 +294,15 @@ const BarListChart: FC< BarListChartProps > = ( {
 	);
 };
 
+// Create composition type
+type BarListChartComponent = FC< BarListChartProps > & {
+	Legend: typeof ChartLegend;
+};
+
+// Attach subcomponents to create composition API
+const BarListChart = attachSubComponents( BarListChartBase, {
+	Legend: ChartLegend,
+} ) as BarListChartComponent;
+
+export { BarListChart };
 export default withResponsive< BarListChartProps >( BarListChart );
```

Comment:
The export doesn't seem consistent with the bar chart:

https://github.com/Automattic/jetpack/pull/44393/files#diff-db6f50e03044a894c48062831a2d8bc57f0670ed456eb10d5c2291e48c8d4551R392